alerts/cluster-monitoring-operator/TargetDown.md

```diff
@@ -0,0 +1,276 @@
+      serverName: node-exporter.openshift-monitoring.svc
+  jobLabel: app.kubernetes.io/name
+  namespaceSelector: {}
+  selector:
+    matchLabels:
+      app.kubernetes.io/component: exporter
+      app.kubernetes.io/name: node-exporter
+      app.kubernetes.io/part-of: openshift-monitoring
+```
+In the `Prometheus` that scrapes this target, the volume mounted at `/etc/prometheus/secrets/*`
+are the secrets having the same name as the subdirectory. In this example, the secret
+is `metrics-client-certs` in the namespace `openshift-monitoring`. Now we extract
+the certificate from the secret using this command:
+```bash
+oc extract secret/$SECRET_NAME -n $NAMESPACE --keys=tls.crt --to=- > certificate.crt
+```
+Then we inspect is expiration date.
+```bash
+openssl x509 -noout -enddate -in certificate.crt
+```
+The output should contain `notAfter` field as its expiration date.
+```bash
+notAfter=Aug  6 13:11:20 2025 GMT
+```
+
+Normally OpenShift **automatically renews** the certificates before expiration date.
+This date should be sometime in the future. If the certificate does expire without
+automatic renewal, please contact OpenShift support team.
+
+If the issue requires immediate resolution, please refer to [this guide on how to
+force a certificate renewal](https://access.redhat.com/solutions/5899121).
+
+To diagnose potential issues with automatic certificate renewal, perform the
+following checks:
+1. Ensure the `prometheus-k8s` stateful set logs do not display errors related to
+   certificates.
+   ```bash
+   oc logs statefulset/prometheus-k8s -n openshift-monitoring
+   ```
+2. Verify that the scraped target pod(s) logs are free from certificate-related errors.
+3. Check that the `cluster-monitoring-operator` pod is running and its logs contain
+   no errors regarding certificate rotation.
+   ```bash
+   oc logs deployment/cluster-monitoring-operator -n openshift-monitoring
+   ```
+4. Confirm the `prometheus-operator` pod is operational and its logs have no errors
+   concerning certificate rotation.
+   ```bash
+   oc logs deployment/prometheus-operator -n openshift-monitoring
+   ```
+
+
+#### Prometheus Scraping Interval
+
+Ensure that the scraping interval for the target is appropriately configured.
+Too frequent scraping can degrade the performance of target pod, even causing
+timeout when scraping.
+
+The last scrape duration is visible on the **Observe** -> **Targets** tab in the
+Openshift web Console.
+
+Otherwise we can check the metric `scrape_duration_seconds`.
+
+If scrape duration is close to the scrape interval, we may consider to increase
+the interval.
+
+The scrape interval is configured in the `ServiceMonitor` or `PodMonitor`.
+
+
+#### Target Metrics Path
+
+Verify that the metrics path in the target is correctly defined in `ServiceMonitor`
+and `PodMonitor`. If due to recent update the target changes the metrics path, the
+metric endpoint should update accordingly.
```

Comment:
```suggestion
Verify that the metrics path in the target is correctly defined in the `ServiceMonitor`
and `PodMonitor` resources. If due to recent update the target changes the metrics path, the
metric endpoint should update accordingly.
```
The last sentence's meaning isn't clear - "due to recent update" of what?

Reply:
here I want to say the scraped target may receive a code update that changes the metrics/ endpoint.